[CS2113-F10-1] CSProjPlanner

WrinklyToad opened the pull request:
Help computer science students (and secondarily students of other majors) who want to better keep track of their projects and school assignments

hearobe committed:
Merge pull request #40 from FTang21/branch-assertions

Some updates

Commit:
Update IO tests

Nineves committed:
Update UserGuide.md

Nineves committed:
Update UserGuide.md

YaxinJoy committed:
Merge pull request #42 from Nineves/master

Add logging and assertions

WrinklyToad committed:
Add test for deleting project

WrinklyToad committed:
add Deadline variable to ProjectListTest

WrinklyToad committed:
Update tests in ProjectTest

WrinklyToad committed:
Rename Duketest to CsProjPlannerTest

Nineves committed:
Merge pull request #43 from johnmcmonigle/branch-Testing

Update JUnit Testing for ProjectListTest & ProjectTest

Commit:
Formatting

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

Commit:
Add more JUnit tests

YaxinJoy committed:
ｒｅｆａｃｔｏｒ　ｔｈｅ　ｃｏｄｅ　ｏｆ　ａｄｄｐｒｏｊｅｃｔ　ａｎｄ　ｄｅｌｅｔｅｐｒｏｊｅｃｔ

YaxinJoy committed:
fix gradlew issue

WrinklyToad committed:
Update contributions

Nineves committed:
Merge pull request #44 from YaxinJoy/master

Refactor the code

WrinklyToad committed:
Update User Guide

Nineves committed:
Merge pull request #45 from johnmcmonigle/branch-UpdateUserGuide

Update user guide

Nineves committed:
Update UserGuide.md

WrinklyToad committed:
Update user guide

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

WrinklyToad committed:
Add UmlDiagrams directory

WrinklyToad committed:
Update Gradle

hearobe committed:
Merge pull request #58 from johnmcmonigle/master

Add UmlDiagrams directory

WrinklyToad committed:
Add UmlDiagrams

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

FTang21 committed:
Merge pull request #59 from johnmcmonigle/master

Add UmlDiagrams

Commit:
NA

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

# Conflicts:
#	src/main/java/cpp/CsProjPlanner.java
#	src/main/java/cpp/projects/commandhandler/CommandHandler.java

Commit:
Update DG

FTang21 committed:
Merge commit '0376d658e2de6a95ff9f2aad6c91501d2184d82b'

* commit '0376d658e2de6a95ff9f2aad6c91501d2184d82b': (21 commits)
  Add UmlDiagrams
  Update Gradle
  Add UmlDiagrams directory
  Update user guide
  Update UserGuide.md
  Update User Guide
  Update contributions
  fix gradlew issue
  ｒｅｆａｃｔｏｒ　ｔｈｅ　ｃｏｄｅ　ｏｆ　ａｄｄｐｒｏｊｅｃｔ　ａｎｄ　ｄｅｌｅｔｅｐｒｏｊｅｃｔ
  Rename Duketest to CsProjPlannerTest
  Update tests in ProjectTest
  add Deadline variable to ProjectListTest
  Add test for deleting project
  Update UserGuide.md
  Update UserGuide.md
  Update IO tests
  Add logging and assertions
  Update UG to contain view command
  Update method names and command names to be consistent
  Update Javadocs to be consistent with each other
  ...

Commit:
Update DG (command parser implementation)

Commit:
Update DG

Commit:
Add command parsers and commands

Commit:
Passed tests

Commit:
text-ui-test

WrinklyToad committed:
Merge pull request #60 from Nineves/master

Update DG
See PR review for approval reasoning

WrinklyToad committed:
Add value proposition

WrinklyToad committed:
Delete 'JohnUmls' file (was a placeholder)

WrinklyToad committed:
Add user stores (v1.0 & v2.0)

WrinklyToad committed:
Add addproject and deleteProject UML diagrams

WrinklyToad committed:
Add addProject and deleteProject to Developer Guide

WrinklyToad committed:
Fix empty return

WrinklyToad committed:
Remove empty return

Nineves committed:
Merge pull request #62 from johnmcmonigle/branch-developerGuide

Update developer guide

Nineves committed:
Update DeveloperGuide.md

Nineves committed:
Update DeveloperGuide.md

hearobe committed:
Add to DeveloperGuide.md

FTang21 committed:
Merge commit '47e893e8abc93f196508a2c7b80a4883fdb3469e'

* commit '47e893e8abc93f196508a2c7b80a4883fdb3469e':
  Add to DeveloperGuide.md
  Update DeveloperGuide.md
  Update DeveloperGuide.md
  Remove empty return
  Fix empty return
  Add addProject and deleteProject to Developer Guide
  Add addproject and deleteProject UML diagrams
  Add user stores (v1.0 & v2.0)
  Delete 'JohnUmls' file (was a placeholder)
  Add value proposition
  text-ui-test
  Passed tests
  Add command parsers and commands
  Update DG
  Update DG (command parser implementation)
  Update DG
  NA
  Add more JUnit tests
  Formatting

FTang21 committed:
Update DG to include AddDeadline implementation

FTang21 committed:
Update AddDeadline.png file

FTang21 committed:
Update AddDeadline.png to not have transparent background

FTang21 committed:
Update AddDeadline.png to AddDeadline.jpg

hearobe committed:
Merge pull request #63 from FTang21/branch-update-DG

Update DG to include AddDeadline implementation

hearobe committed:
Add function to list programming languages

WrinklyToad committed:
Merge branch 'branch-developerGuide'

hearobe committed:
Add function to add programming languages to a project

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

YaxinJoy committed:
add save and load based on current structure; update developer guide.

YaxinJoy committed:
Fix gradlew issue.

Commit:
Updated command and command parsers

WrinklyToad committed:
Merge pull request #66 from YaxinJoy/master

add save and load based on current structure and update developer guide.

WrinklyToad committed:
Merge branch 'master' into master

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

# Conflicts:
#	src/main/java/cpp/model/project/Todo.java

Commit:
fixed conflict

WrinklyToad committed:
Merge pull request #68 from Nineves/master

Updated command and command parsers

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

FTang21 committed:
Create Deadline class

Deadline currently understands two formats
- yyyy/mm/dd
- a given day of the week

WrinklyToad committed:
Add new UML diagrams and more content to Developer Guide

FTang21 committed:
Update CommandHandler to contain AddDeadlineCommand

FTang21 committed:
Update Minor details on AddDeadline command

FTang21 committed:
Update dayAparts to be 7 days for if the days are equal

FTang21 committed:
Merge pull request #69 from johnmcmonigle/branch-DeveloperGuide

Add more UML Diagrams for adding/deleting projects & update developer guide

FTang21 committed:
Fix formatting errors

FTang21 committed:
Update junit test

FTang21 committed:
Update junit test for proper dates

WrinklyToad committed:
Merge pull request #70 from FTang21/branch-update-Deadline

Update Deadline into a class

WrinklyToad committed:
Update tests

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp into branch-Tests

FTang21 committed:
Update UserGuide.md and error response to include format

WrinklyToad committed:
Add all Command Class tests

Nineves committed:
Merge pull request #71 from johnmcmonigle/branch-CommandClassTests

Add tests for all Command classes

WrinklyToad committed:
Add tests for all Parser classes

WrinklyToad committed:
Fix JavaDoc problem

YaxinJoy committed:
Update UserGuide.md

Commit:
Updated User Guide

WrinklyToad committed:
Fix Another JavaDoc issue

Commit:
fixed small problems

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

Nineves committed:
Merge pull request #72 from johnmcmonigle/branch-ParserClassTest

Add tests for all Parser classes

hearobe committed:
Merge from master

hearobe committed:
Change language methods to use parser and commands

hearobe committed:
Merge branch 'master' into list-programming-languages

hearobe committed:
Merge branch 'master' of https://github.com/Nineves/tp into Nineves-master

hearobe committed:
Merge branch 'Nineves-master'

tanweili commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+
+####Delete a Project
+![image info](./UmlDiagrams/deleteProjectNew.png)
+*The steps are omitted here as they are the exact same as adding a project, only with the differing class names previously mentioned.
```
Should add a line break here. The picture and the words start on the same line on my screen. Or somehow make the sentence start on the new line.

tanweili commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+
+**Step 5.** If the index of the project was found, call that specific Project’s `setDeadline()` method and update the deadline.
+
+![image info](./UmlDiagrams/AddDeadline.jpg)
```
It would be great to show to show a user icon to explicitly show that it is the user calling the start of the sequence diagrams, though I believe this might not be absolutely necessary.

ivanaitzliddat replied:
1. Possible to add square braces for conditions in opt box
![image](https://user-images.githubusercontent.com/69310031/160964368-fb9b3642-cd70-4f27-94a1-486f8aa822bf.png)

tanweili commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+Because of this, many sequence diagrams look similar in the current version. The following sequence diagrams and explanations show what some of our features looked like at the end of v1.0:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProject.png)
```
Is the floating return statement adhering to the UML standards?

ivanaitzliddat commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+The command to add a project is one example of this structure:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProjectNew.png)
```
1. Possible to remove the arrow that has the name "return" on it
![image](https://user-images.githubusercontent.com/69310031/160963240-0064cdaf-92e9-4c96-8ded-07bfec33f13b.png)

2. The arrow pointing from `Command Handler` self-invocation activation bar should touch the activation bar of `DeleteProjectCommand`.
![image](https://user-images.githubusercontent.com/69310031/160963261-39bf6aac-6a29-4f26-b3ab-9dd91d5de062.png)

suenalaba replied:
I second this change. The return arrow should be removed. Furthermore, I think some diagramming tool can be used to make the lines straighter.

ivanaitzliddat commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+#### Add a Project
+![image info](./UmlDiagrams/addProjectNew.png)
+
+**Step1.** After recieving the necessary method call, the `CommandHandler` class will self-call executeCommand with both the list of projects as well as a constructor call to AddProjectCommandParser (and a call to the parse method of AddProjectCommandParser) as its parameters
```
Perhaps can try to use `AddProjectCommandParser#parse` to represent the calling of parse method

tanweili commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+*The steps are omitted here as they are the exact same as adding a project, only with the differing class names previously mentioned.
+
+#### Print Project List
+![image info](./UmlDiagrams/printProject.png)
```
Perhaps the opt frame can be a alt frame instead?

SeanHoWB replied:
i second this

ivanaitzliddat commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+Deleting a project will follow nearly the exact same structure; the only difference is that the corresponding Parser and Command objects will be used (`DeleteProjectCommandParser` and `DeleteProjectCommand`)
+
+####Delete a Project
+![image info](./UmlDiagrams/deleteProjectNew.png)
```
Same comments as the sequence diagram above

![image](https://user-images.githubusercontent.com/69310031/160963337-77407d74-332c-43f3-86fb-91f9c4b599f9.png)

![image](https://user-images.githubusercontent.com/69310031/160963357-4cecef7f-5063-4a1a-95dd-c3509b370647.png)

suenalaba replied:
Shouldn't arrow head touch the tip of the activation bar. Furthermore, shouldn't the floating arrows be removed?

tanweili reviewed (commented):
Overall I think that the DG is well documented thus far with many details regarding the different classes. Perhaps some UML diagrams details can be refined.

ivanaitzliddat commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+*The steps are omitted here as they are the exact same as adding a project, only with the differing class names previously mentioned.
+
+#### Print Project List
+![image info](./UmlDiagrams/printProject.png)
```
1. The `deadline` returned from `Project` object should perhaps be at the end of the activation bar
![image](https://user-images.githubusercontent.com/69310031/160963463-70ecc875-4fe7-44a8-9e59-9b3ff8dcb556.png)

2. It would be good to ensure that all the arrows (that are relevant) start and end at the edge of the activation bars
![image](https://user-images.githubusercontent.com/69310031/160963488-3df4b62a-9aff-4c8e-a2b8-f54d41f090e7.png)


3. Should there be a return from self-invocation of `CommandHandler`? (Not compulsory but just in case it returns something or so)
![image](https://user-images.githubusercontent.com/69310031/160963517-a57c0d74-9818-4fc6-9a9e-79623ed3c14b.png)

ivanaitzliddat commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+**Step5.** the general information of a project, i.e., its title and deadline, will be displayed.
+
+#### Todo feature
+![image info](./UmlDiagrams/Todo.png)  
```
1. Perhaps the return arrow can touch the activation bar of the self-invocation to make it less ambiguous
![image](https://user-images.githubusercontent.com/69310031/160963650-2d7dd078-6511-48cc-8124-4191872b4c5f.png)

2. targetProject returned can possibly add the type and make sure that the arrow is end of the activation bar
![image](https://user-images.githubusercontent.com/69310031/160963789-d18902e7-9334-44ad-9d1a-93ab7db1e464.png)

3. Should there be a return arrow from the self-invocation activation bar?
![image](https://user-images.githubusercontent.com/69310031/160963844-b56ab941-734f-4cef-95fc-5cca92a90d56.png)

4. Possible to remove the italics from close bracket of the condition in the opt box (Should there be a return arow as well?)
![image](https://user-images.githubusercontent.com/69310031/160963923-a5f68089-23ec-4d1b-a4dd-a6a2da60a4df.png)

SeanHoWB commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+The command to add a project is one example of this structure:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProjectNew.png)
```
I strongly suggest using plantuml, the image is hard to read with different font size, 
![image](https://user-images.githubusercontent.com/69450377/160964004-a7fffa4e-b066-481b-aa5f-6b6346426c34.png)
different arrows
![image](https://user-images.githubusercontent.com/69450377/160964074-f2ea2b95-9b37-413c-9aa1-03a32fe304e1.png)
floating returns
![image](https://user-images.githubusercontent.com/69450377/160964100-bd8c8247-fa34-4822-94da-f1ced7ee0bec.png)
lack of colour...

Really plantuml helps solve all of that, i suggest you try

SeanHoWB commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+Deleting a project will follow nearly the exact same structure; the only difference is that the corresponding Parser and Command objects will be used (`DeleteProjectCommandParser` and `DeleteProjectCommand`)
+
+####Delete a Project
+![image info](./UmlDiagrams/deleteProjectNew.png)
```
![image](https://user-images.githubusercontent.com/69450377/160964276-40bdfc66-e2df-469f-9f37-1aa844b162a1.png)
arrow need to touch top of activation bar

SeanHoWB commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+
+Deleting a project will follow nearly the exact same structure; the only difference is that the corresponding Parser and Command objects will be used (`DeleteProjectCommandParser` and `DeleteProjectCommand`)
+
+####Delete a Project
```
```suggestion
#### Delete a Project
```

ivanaitzliddat commented on docs/DeveloperGuide.md:
````diff
@@ -6,32 +6,215 @@
+
+**Step 6.** ```printDetails()``` displays project title and deadline to the user. It iterates through the ```Todo```s in the ```todos``` attribute and prints each `Todo` by calling `Todo`’s `toString()` function
+
+![image info](./UmlDiagrams/ViewProject.jpg)
````
1. Consider using opt rather than alt for this as there is only 1 condition (Condition can also be placed at the top of the box)
![image](https://user-images.githubusercontent.com/69310031/160964429-bc907229-b443-4482-9fb8-a8ef0e93bdf7.png)

ivanaitzliddat replied:
This seems to be used by a different software as compared to other diagrams

SeanHoWB commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+![image info](./UmlDiagrams/deleteProjectNew.png)
+*The steps are omitted here as they are the exact same as adding a project, only with the differing class names previously mentioned.
+
+#### Print Project List
```
![image](https://user-images.githubusercontent.com/69450377/160964582-49152343-2278-476d-a49e-720d9b7e8a8d.png)
need arrow to deactivate

SeanHoWB reviewed (commented):
overall standardise on the colour, and use plantuml for better diagrams, ATB!

ivanaitzliddat commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+**Step4.** This new `Project` object is returned to the ProjectList, added, and the output "[projectName] added." is given to the user.
+
+#### Delete a Project
+![image info](./UmlDiagrams/deleteProject.png)
```
1. Possible to use software all the way rather than adding handdrawn lines
![image](https://user-images.githubusercontent.com/69310031/160965519-b1ead20b-3fdd-4de4-b123-2f5aa860dc52.png)

2. Consider removing the circle at the start of the arrow
![image](https://user-images.githubusercontent.com/69310031/160965554-140a08da-1bec-4a28-91f1-b9a13b38461b.png)

ivanaitzliddat commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
+
+**Step5.** The `Project` object will create a `Todo` object and add it to the `ArrayList<Todo>` in `Project` class.  
+
+#### View a Project
```
Would be good if you all can decide whether to put images before steps or steps before images

ivanaitzliddat commented on docs/DeveloperGuide.md:
```diff
@@ -6,32 +6,215 @@
 ### Target user profile
 
-{Describe the target user profile}
+* computer science university students
+* has a need to manage a significant number of projects and tasks
+* can type fast
+* prefers typing to mouse interactions
+* is reasonably comfortable using CLI apps
```
Possible to have better formatting for this portion

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

Commit:
change package name

WrinklyToad committed:
Merge pull request #73 from Nineves/master

Updated Developer Guide

Nineves committed:
Update DeveloperGuide.md

Nineves committed:
Update DeveloperGuide.md

WrinklyToad committed:
Update Project.java to hold links (as Strings)

FTang21 committed:
Merge commit 'b7694e810afc6c68de79cfbba5da0b79382470e7' into branch-update-Deadline

* commit 'b7694e810afc6c68de79cfbba5da0b79382470e7':
  Update DeveloperGuide.md
  Update DeveloperGuide.md
  change package name
  fixed small problems
  Fix Another JavaDoc issue
  Updated User Guide
  Update UserGuide.md
  Fix JavaDoc problem
  Add tests for all Parser classes
  Add all Command Class tests
  Update tests
  Add new UML diagrams and more content to Developer Guide

FTang21 committed:
Update addDeadline to addProjectDeadline

Change for upcoming addTodoDeadline

hearobe committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

hearobe committed:
Merge branch 'master' into list-programming-languages

WrinklyToad committed:
Add ability for user to change GitHub link repo for a project

hearobe committed:
fixed package structure errors

Nineves committed:
Merge pull request #74 from AY2122S2-CS2113-F10-1/list-programming-languages

List programming languages

FTang21 committed:
Add new command addTodoDeadline

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp into branch-ReposAndLinks

# Conflicts:
#	src/main/java/cpp/logic/CommandHandler.java
#	src/main/java/cpp/model/project/Project.java
#	src/main/java/cpp/ui/Constants.java

FTang21 committed:
Merge branch 'master' into branch-update-Deadline

* master:
  fixed package structure errors
  Change language methods to use parser and commands
  Add function to add programming languages to a project
  Add function to list programming languages

# Conflicts:
#	src/main/java/cpp/logic/CommandHandler.java
#	src/main/java/cpp/model/ProjectList.java
#	src/main/java/cpp/model/project/Project.java
#	src/main/java/cpp/ui/Constants.java

FTang21 committed:
Edit minor updates

FTang21 committed:
Update JUnit tests

FTang21 committed:
Update Format on test

FTang21 committed:
Update testDeadline()

FTang21 committed:
Update formatting on testDeadline()

Nineves committed:
Merge pull request #75 from FTang21/branch-update-Deadline

Added Task deadlines

WrinklyToad committed:
Allow user to open GitHub repo

WrinklyToad committed:
Allow reading and writing of git URL to the disk

WrinklyToad committed:
Update Storage to use ` instead of semi-colons

Reasoning being that backticks cannot be in URLs. Before, if the user input a URL with a semi-colon, it would cause an error.

WrinklyToad committed:
Merge branch 'master' into branch-ReposAndLinks

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp into branch-ReposAndLinks

# Conflicts:
#	src/data/projectList.txt
#	src/main/java/cpp/logic/CommandHandler.java
#	src/main/java/cpp/model/ProjectList.java
#	src/main/java/cpp/ui/Response.java

WrinklyToad committed:
Fix storage conflict

WrinklyToad committed:
Merge branch 'branch-ReposAndLinks' of https://github.com/johnmcmonigle/tp into branch-ReposAndLinks

# Conflicts:
#	src/data/projectList.txt
#	src/main/java/cpp/logic/CommandHandler.java
#	src/main/java/cpp/model/ProjectList.java

WrinklyToad committed:
Fix JavaDoc comments

WrinklyToad committed:
Fix Javadoc error

WrinklyToad committed:
Re-write projectList.txt to (hopefully) fix build error

WrinklyToad committed:
Fix problem with storage

WrinklyToad committed:
Fix (hopefully, please) the build error

WrinklyToad committed:
Fix test case

WrinklyToad committed:
Fix (please) directory issue

WrinklyToad committed:
Empty projectList.txt

WrinklyToad committed:
Remove print statement and fix math error on storage

WrinklyToad committed:
Update test case

WrinklyToad committed:
FIx test case

WrinklyToad committed:
Add git commands to user guide

Nineves committed:
Merge pull request #76 from johnmcmonigle/branch-ReposAndLinks

Add ability for user to change and open the GitHub repo of the project

FTang21 committed:
Update DeveloperGuide for Deadline commands

FTang21 committed:
Update UserGuide

Commit:
Updated UG&DG, added some comments

WrinklyToad committed:
Merge pull request #77 from FTang21/branch-update-Deadline

Update DeveloperGuide for deadline commands

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

# Conflicts:
#	docs/UserGuide.md

Commit:
delete add web links

Commit:
Add ending period

FTang21 committed:
Merge pull request #78 from Nineves/master

Updated UG&DG, added some comments

Nineves committed:
Update UserGuide.md

WrinklyToad committed:
Remove Comments and add projectList.txt to gitignore

Commit:
The app is able to store languages for each project

WrinklyToad committed:
Fixed empty catch block

FTang21 committed:
Merge pull request #80 from Nineves/master

The app is able to store languages for each project

Nineves committed:
Merge pull request #79 from johnmcmonigle/branch-MinorFixes

Remove Comments and add projectList.txt to gitignore

Nineves committed:
Update UserGuide.md

Nineves committed:
Update UserGuide.md

Nineves committed:
Update UserGuide.md

YaxinJoy committed:
Set theme jekyll-theme-slate

YaxinJoy committed:
Set theme jekyll-theme-dinky

LI-XINYI1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,293 @@
+
+
+**Model Component**
+![image info](./UmlDiagrams/Model.png)
```
Just for a better viewing experience, maybe you can use horizontal and vertical lines to connect the boxes of classs objects?

LI-XINYI1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,293 @@
+
+The basic flow of the structure:
+
+![image info](./UmlDiagrams/CommandParser.png)
```
For the sequence diagram, can add object deletion notation when the TodoCommand is not in use, though it is automatically managed by java but can still add it for clearity.

suenalaba replied:
For the sequence diagram for command parser, shouldn't the activation box for :AddProjectCommandParser end once the reply message ends?

LI-XINYI1 commented on docs/DeveloperGuide.md:
````diff
@@ -1,37 +1,293 @@
+**Step 6.** ```printDetails()``` displays project title and deadline to the user. It iterates through the ```Todo```s in the ```todos``` attribute and prints each `Todo` by calling `Todo`’s `toString()` function
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![image info](./UmlDiagrams/ViewProject.jpg)
````
It might be better to put the [condition statement] on the right of the "alt" box instead of putting it on top of the arrow of the returning method.
![image](https://user-images.githubusercontent.com/72705487/161185072-70df8df0-e832-44b9-9976-11bfd1c9b42e.png)

Similar changes can be applied to the two loop sections.

suenalaba replied:
Shouldn't arrow heads be at the tip of the activation bars?
Also, for the alt statement, wouldn't it be better to use an opt since there is no else clause?

LI-XINYI1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,293 @@
+Because of this, many sequence diagrams look similar in the current version. The following sequence diagrams and explanations show what some of our features looked like at the end of v1.0:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProject.png)
```
Will it be better you enlarge the "opt: no project name provoided" to cover the commandHandler bar, as it is a part of condition check?

suenalaba commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,293 @@
+*The steps are omitted here as they are the exact same as adding a project, only with the differing class names previously mentioned.
+
+#### Print Project List
+![image info](./UmlDiagrams/printProject.png)
```
I think that it would be better for the opt frame to be alt since there is an else statement?

suenalaba commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,293 @@
+
+**Step 9.** Inside the constructor it considers 2 types of inputs, a day of the week or a date format of yyyy-mm-dd. If it is a day of the week, it will properly detail next day that day of the week from the current day.
+
+![image info](./UmlDiagrams/AddProjectDeadline.jpg)
```
The arrow heads are exceeding the activation bars. The text is abit hard to view..

suenalaba commented on docs/DeveloperGuide.md:
```diff
@@ -1,37 +1,293 @@
+
+**Step 9.** Inside the constructor it considers 2 types of inputs, a day of the week or a date format of yyyy-mm-dd. If it is a day of the week, it will properly detail next day that day of the week from the current day.
+
+![image info](./UmlDiagrams/AddTodoDeadline.jpg)
```
Add todo deadline command object is overrlapping with the activation bar.

suenalaba reviewed (commented):
Try to standardise diagramming tools used.

Commit:
Some changes on UG, DG

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

Nineves committed:
Set theme jekyll-theme-minimal

Nineves committed:
Update UserGuide.md

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

Nineves committed:
Update DeveloperGuide.md

fixed hyperlink issues

Commit:
update DG

Nineves committed:
Update UserGuide.md

Nineves committed:
Update UserGuide.md

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

# Conflicts:
#	docs/DeveloperGuide.md

Commit:
Fixed many bugs

Commit:
Passed tests

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

# Conflicts:
#	docs/UserGuide.md

Commit:
no message

Commit:
closes #124

WrinklyToad committed:
Merge pull request #133 from Nineves/master

Solved some issues

FTang21 committed:
Update commands and UI classes

Changed remove `add` from deadline commands
Change certain commands from project_name to project_index
Update to UI and handling of IndexOutOfBoundsExceptions

FTang21 committed:
Update UI tests

FTang21 committed:
Update JUnit test and remove redundant code

FTang21 committed:
Fix checkstyle error

FTang21 committed:
Update UserGuide to keep format consistent

Closes #127

FTang21 committed:
Update deadline UI to 'set' instead of 'add'

Also forces help to be a single argument
Closes #92

FTang21 committed:
Fix checkstyle error

WrinklyToad committed:
Merge pull request #134 from FTang21/branch-PED

Updated some stuff

Commit:
updated developer guide

Commit:
updated PPP

Commit:
update PPP

WrinklyToad committed:
Merge pull request #135 from Nineves/master

Updated DG (added manual testing instructions) & updated PPP

YaxinJoy committed:
Update based on code review

YaxinJoy committed:
Merge pull request #136 from YaxinJoy/master

Update based on code review

YaxinJoy committed:
Update diagram in DG based on review

YaxinJoy committed:
Update UG and PPP.

YaxinJoy committed:
Fix assertion error

YaxinJoy committed:
Merge pull request #137 from YaxinJoy/master

Several updates based on review

FTang21 committed:
Updated DG and PPP

hearobe committed:
updated ppp

WrinklyToad committed:
Fix coding standard violation

WrinklyToad committed:
Extract out magic constants

FTang21 committed:
Add new JUnit tests to increase code coverage

FTang21 committed:
Merge branch 'master' into branch-PED

* master:
  updated ppp
  Fix assertion error
  Update UG and PPP.
  Update diagram in DG based on review

# Conflicts:
#	src/main/java/cpp/model/ProjectList.java

FTang21 committed:
Fix checkstyle errors

FTang21 committed:
Fix checkstyle errors

FTang21 committed:
Add debug test

FTang21 committed:
Update CsProjPlannerTest

FTang21 committed:
Remove failing test and Add testInput for coverage

WrinklyToad committed:
Merge pull request #138 from FTang21/branch-PED

Update to Documentation

WrinklyToad committed:
Change opengit command to use project index instead of name

WrinklyToad committed:
Fix merge issue

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp into branch-FixIssues

WrinklyToad committed:
Change view command to use project index instead of name

WrinklyToad committed:
Fix userguide and test error

WrinklyToad committed:
Change listlanguages command to use project index instead of project name

WrinklyToad committed:
Fix test cases

WrinklyToad committed:
Update add/delete project diagrams

WrinklyToad committed:
Update sequence diagrams and add openGit and changeGit to DG

WrinklyToad committed:
Add note to UG

WrinklyToad committed:
Fix indentation

WrinklyToad committed:
Update test case

WrinklyToad committed:
Fix test case

WrinklyToad committed:
Empty Project List (developer copy)

WrinklyToad committed:
Empty projectList (local copy)

WrinklyToad committed:
Revert last commit

WrinklyToad committed:
Fix test case (hopefully)

WrinklyToad committed:
Fix test case

WrinklyToad committed:
Fix test case

WrinklyToad committed:
Fix test case for batch file

Nineves committed:
Merge pull request #139 from johnmcmonigle/branch-FixIssues

Fix project_index/title inconsistencies, update DG and UG, various fixes

Nineves committed:
Update AboutUs.md

Nineves committed:
Update AboutUs.md

Nineves committed:
Update AboutUs.md

Nineves committed:
Update AboutUs.md

Nineves committed:
Set theme jekyll-theme-cayman

Nineves committed:
Update DeveloperGuide.md

FTang21 committed:
Update DG images and AboutUs image

YaxinJoy committed:
Merge pull request #140 from FTang21/master

Update DG images and AboutUs image

YaxinJoy committed:
Rename WanYaxin.md to yaxinjoy.md

WrinklyToad committed:
Merge branch 'branch-FixIssues'

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

WrinklyToad committed:
Update PPP and add two links to DG

hearobe committed:
removed unnecessary comments, made magic strings constants

WrinklyToad committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp into branch-PPP

hearobe committed:
Merge pull request #141 from johnmcmonigle/branch-PPP

Update PPP

WrinklyToad committed:
Fix UG links

WrinklyToad committed:
Fix DG link & table

WrinklyToad committed:
Added all PPPs to home page

Nineves committed:
Merge pull request #142 from johnmcmonigle/branch-PPP

Add PPP pages & Fix DG links & UG table

Commit:
NA

Commit:
update DG

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-1/tp

# Conflicts:
#	docs/DeveloperGuide.md
#	src/main/java/cpp/ui/Constants.java

Commit:
update DG

YaxinJoy committed:
Merge pull request #143 from Nineves/master

Update the DG according to our latest version

Nineves committed:
Update DeveloperGuide.md

Nineves committed:
Update UserGuide.md
